src/activities/components/ActivityCreateButton.vue

```diff
@@ -0,0 +1,238 @@
+          </template>
+          <template #selected-item="scope">
+            <div class="row no-wrap ellipsis">
+              <QIcon
+                :name="scope.opt.icon"
+                size="1.1em"
+                class="on-left q-ml-xs"
+                color="positive"
+              />
+              <div class="ellipsis">
+                {{ scope.opt.label }}
+              </div>
+            </div>
+          </template>
+        </QSelect>
+        <QOptionGroup
+          v-model="isSeries"
+          :options="[
+            { label: $t('CREATEACTIVITY.ONE_TIME_ACTIVITY'), value: false },
+            { label: $t('ACTIVITYMANAGE.SERIES'), value: true },
+          ]"
+          color="primary"
+          inline
+          class="q-mb-md"
+        />
+      </Component>
+    </div>
+  </QDialog>
+</template>
+
+<script setup>
+import addHours from 'date-fns/addHours'
+import addSeconds from 'date-fns/addSeconds'
+import startOfTomorrow from 'date-fns/startOfTomorrow'
+import {
+  QItem,
+  QItemSection,
+  QItemLabel,
+  QIcon,
+  QFab,
+  QFabAction,
+  QDialog,
+  QSelect,
+  QOptionGroup,
+} from 'quasar'
+import { ref, computed, watch, unref, defineAsyncComponent } from 'vue'
+
+import { useActivityTypeHelpers } from '@/activities/helpers'
+import { useCreateActivityMutation, useCreateActivitySeriesMutation } from '@/activities/mutations'
+import { useActivityTypeService } from '@/activities/services'
+import { defaultDuration } from '@/activities/settings'
+import { useCurrentGroupService } from '@/group/services'
+import { usePlaceService, usePlaceTypeService } from '@/places/services'
+
+const ActivityEdit = defineAsyncComponent(() => import('./ActivityEdit.vue'))
+const ActivitySeriesEdit = defineAsyncComponent(() => import('./ActivitySeriesEdit.vue'))
+
+const isOpen = ref(false)
+const placeId = ref(null)
+const isSeries = ref(false)
+
+const newActivity = ref({})
+const newSeries = ref({})
+
+const {
+  groupId,
+} = useCurrentGroupService()
+
+const {
+  getPlacesByGroup,
+} = usePlaceService()
+
+const {
+  getPlaceTypeById,
+} = usePlaceTypeService()
+
+const places = computed(() => getPlacesByGroup(groupId).filter(place => place.status === 'active'))
+
+const {
+  mutate: saveNewActivity,
+  status: createActivityStatus,
+  reset: resetNewActivity,
+} = useCreateActivityMutation({
+  onSuccess () {
+    isOpen.value = false
+  },
+})
+
+const {
+  mutate: saveNewSeries,
+  reset: resetNewSeries,
+  status: createSeriesStatus,
+} = useCreateActivitySeriesMutation({
+  onSuccess () {
+    isOpen.value = false
+  },
+})
+
+const {
+  getIconProps,
+} = useActivityTypeHelpers()
+
+const {
+  getActivityTypesByGroup,
+} = useActivityTypeService()
+
+const activityTypes = computed(() => getActivityTypesByGroup(groupId, { status: 'active' }))
+
+watch(placeId, id => {
+  // replace object to trigger 'value' watcher in editMixin
+  newActivity.value = {
+    ...newActivity.value,
+    place: unref(id),
```

Comment:
Also, I'm not sure why setting the `place` attribute on the `newActivity` object isn't triggering watchers anyway...?

Could the `v-model` in the place select just be:
```html
v-model="(isSeries ? newSeries : newActivity).place"
```

Reply:
I was stuck at this for some time, I think the `value` watcher in `editMixin` is shallow (it updates `this.edit´ when `this.value` changes), so it needs a new object to make it work. I could also make it a deep watch, but I found it simpler to just replace the object.

Reply:
Ah yeah, I could imagine it being shallow. As the diff checking is deep (deepEqual) I think I somehow imagined in my head the watching was, because in reality the `value` is not changing... as it's just the initial value... oh, which actually makes me wonder why we need to change the value, not just the edit part. But my head is not deep in there, so whatever works :)